packages/uni_app/lib/view/schedule/schedule.dart

```diff
@@ -67,7 +67,34 @@ class SchedulePageViewState extends State<SchedulePageView>
       length: 6,
     );
 
-    final weekDay = widget.currentWeek.start.weekday;
+    var weekDay = widget.currentWeek.start.weekday;
+
+    final lecturesThisWeek = <Lecture>[];
+    widget.currentWeek.weekdays.take(6).forEach((day) {
+      final lectures = lecturesOfDay(widget.lectures, day);
+      lecturesThisWeek.addAll(lectures);
+    });
```

Comment:
Since this code is repeated, I'll propose a refactoring to help cleanup the code.

First thing: since we only need the lectures for the current day, there is no need for the widget to get all of the lectures.

*To be continued...*

Reply:
Change these two lines (https://github.com/NIAEFEUP/uni/blob/fix%2Fschedule-card-redirect/packages%2Funi_app%2Flib%2Fview%2Fschedule%2Fschedule.dart#L16-L18) to:

```
  SchedulePage({super.key, DateTime? now}) : week = Week(now: now ?? DateTime.now());

  final Week week;
```

Reply:
You can store the lecturesThisWeek at the state level (by creating a property `late final List<Lecture> lecturesThisWeek`) and setting it in the initState method.

Then, below, you can simply access that array instead of filtering again.

Reply:
The repeated code actually does slightly different things. The first one is to get the lectures of the week (using `forEach`) and the other to map the days of the week to the lectures of each day (using `map`). 

I made changes to the `build` function to look for the lectures of the day on the `lecturesThisWeek` list instead of going through the full list of lectures of the semester for each day of the week.

Another alternative could be to initially store `lecturesThisWeek` in a `Map` and the build code can iterate only once over this map to build the widget list. What do you think?